common/db.go

```diff
@@ -288,7 +303,47 @@ func (db *DB) SelectOrchs() ([]*DBOrch, error) {
 	rows, err := db.selectOrchs.Query()
 	defer rows.Close()
 	if err != nil {
-		glog.Error("db: Unable to get orchestrators updated in the last 24 hours", err)
+		glog.Error("db: Unable to get orchestrators updated in the last 24 hours: ", err)
+		return nil, err
+	}
+	orchs := []*DBOrch{}
+	for rows.Next() {
+		var orch DBOrch
+		var serviceURI string
+		var ethereumAddr string
+		if err := rows.Scan(&serviceURI, &ethereumAddr); err != nil {
+			glog.Error("db: Unable to fetch orchestrator ", err)
+			continue
+		}
+		orch.ServiceURI = serviceURI
+		orch.EthereumAddr = ethereumAddr
+		orchs = append(orchs, &orch)
+	}
+	return orchs, nil
+}
+
+func (db *DB) FilterOrchs(maxPrice *big.Rat) ([]*DBOrch, error) {
```

Comment:
We could have. I think semantically FilterOrchs represents better what it does, since it now takes in a Filter struct rather than just a price (for possible flexibility towards the future). I thought this would have been cleaner / clearer than changing every selectOrchs statement to take in nil/empty filter struct, I haven't figured out yet if that's me being lazy or not 😅 .

Reply:
> I thought this would have been cleaner / clearer than changing every selectOrchs statement to take in nil/empty filter struct,

We could avoid this while also minimizing code duplication a bit by extracting the boilerplate code into a helper function.

Ex.

```golang
func (db *DB) SelectOrchs() ([]*DBOrch, error) {
    // Other logic

    return db.selectOrchs(DBOrchFilter{})
}

func (db *DB) FilterOrchs(filter DBOrchFilter) ([]*DBOrch, error) {
    // Other logic

    return db.selectOrchs(filter)
}

func (db *DB) selectOrchs(filter DBOrchFilter) ([]*DBOrch, error) {
    var (
        rows *sql.Rows
        err error
    )

    if filter == DBOrchFilter{} {
        rows, err = db.selectOrchs.Query()
    } else {
        var fpMaxPrice int64
        // Convert maxPrice -> fpMaxPrice
        rows, err = db.filterOrchs.Query(fpMaxPrice)
    }
    
    // Error handling, row parsing, etc.
    // Return orchs
}
```

Reply:
In practice, we won't be using `SelectOrchs()` anywhere else (other than testing) so I have a preference for keeping things lean and removing unused code. Using `SelectOrchs(filter)` is clear enough to me, but I'll leave that up to you.

Reply:
Just out of curiosity. What's the idiomatic way?

A. Using a value arg and passing in an empty struct when no filter is used 
B. Using a pointer and passing in nil when no filter is used 

I'm thinking passing a value is okay since we're not modifying the underlying struct and the struct is not going to be very large in the foreseeable future? But passing in an empty struct as argument seems 'dirtier' than passing in nil although that might just be visually?

Reply:
Structs should typically be passed in as a pointer (option A) so we don't incur the cost of copying the entire struct each time the parameter is called.

At least, that's how it is in most languages with some control over addressing; not sure if golang's semantics are different in that regard.

Reply:
Let me repost my comment after second though. 

What is the idiomatic go way? 

A. Pointer and passing in nil when filter is not being used 
B. Value and empty struct when filter is not used 

I often read if it's a struct pass a pointer but that's just not a good argument I think. 
I mean we're not modifying the struct and it isn't very large either so a passing it as a value is okay and using the stack means it doesn't have to be garbage collected. 
It's just that visually being able to pass `nil` is a bit neater? 

Preferences?

Reply:
Even if the struct is small now, we don't know if it'll always be small. In the interests of API stability, I'd go with option A.

But I'd also be totally fine with reverting to the original approach of passing in a `big.Rat` directly, because it's still unclear how we'd compose various config options, and there are some aspects that could be "configurable" on paper but aren't right now (filtering by date).

Reply:
Gotcha, thanks.